File: 0015-3sum/0015-3sum.py
class Solution(object):
    def threeSum(self, nums):
        nums.sort()
        set1=set()
        for i in range(len(nums)-2):
            left=i+1
            right=len(nums)-1
            while left<right:
                total=nums[i]+nums[left]+nums[right]
                if total==0:
                    triplet=tuple([nums[i],nums[left],nums[right]])
                    set1.add(triplet)
                    left += 1
                    right -= 1
                elif total<0:
                    left+=1
                elif total>0:
                    right-=1    

        return list(set1)       
                        
        

File: 0015-3sum/0015-3sum.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());

        set<vector<int>> set1;

        for (int i = 0; i < nums.size() - 2; i++) {
            int left = i + 1;
            int right = nums.size() - 1;

            while (left < right) {
                int total = nums[i] + nums[left] + nums[right];

                if (total == 0) {
                    vector<int> triplet = {
                        nums[i],
                        nums[left],
                        nums[right]
                    };

                    set1.insert(triplet);

                    left++;
                    right--;
                }
                else if (total < 0) {
                    left++;
                }
                else {
                    right--;
                }
            }
        }

        vector<vector<int>> result(set1.begin(), set1.end());

        return result;
    }
};